### ISSUE TITLE
False positive of `font-family` SCSS variables for `value-keyword-case`

### ISSUE THREAD

**Issue by AlexWayfer** [stylelint/stylelint#4622]
> Clearly describe the bug

There are false positives for `font-family` value of SCSS variable.

> Which rule, if any, is the bug related to?

`value-keyword-case`

> What code is needed to reproduce the bug?

```scss
$main_font_family: -apple-system, BlinkMacSystemFont, sans-serif;
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "value-keyword-case": "lower"
  }
}
```

> Which version of stylelint are you using?

`13.2.0` (and online demo)

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI and Atom Plugin.

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to SCSS variables.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
Expected "BlinkMacSystemFont" to be "blinkmacsystemfont" (value-keyword-case)
```

**Comment by jeddy3** [comment]
@AlexWayfer Thanks for the report and for using the template.

Rules are strict by default but you can make them more permissive by configuring a rule's optional secondary option. From the [rule's README](https://github.com/stylelint/stylelint/tree/master/lib/rules/value-keyword-case):

> This rule ignores [`<custom-idents>`](https://developer.mozilla.org/en/docs/Web/CSS/custom-ident) __of known properties__. Values which are paired with non-properties (e.g. `$vars` and custom properties), and do not conform to the primary option, can be ignored using the `ignoreValues: []` secondary option. [emphasis mine]

I suspect that users will want to enforce a case for most variables (including custom properties), which is why the rule doesn't ignore them by default. As variables aren't known properties, stylelint can't know if a variable might have custom-idents in its value.

You can use to use either the `ignoreValues` or `ignoreProperties` secondary options.

For example:

```json
{
  "rules": { "value-keyword-case": [true, { "ignoreValues": ["BlinkMacSystemFont"] }] }
}
```

Or

```json
{
  "rules": { "value-keyword-case": [true, { "ignoreProperties": ["$main_font_family"] }] }
}
```

Or, you can ignore all variables using regex:

```json
{
  "rules": { "value-keyword-case": [true, { "ignoreProperties": ["/^$/"] }] }
}
```

**Comment by AlexWayfer** [comment]
@jeddy3 Thank you!

**Comment by AlexWayfer** [comment]
Just notes:

1. `value-keyword-case` can't accept `true`, it accepts `"lower"|"upper"`
2. There are no `ignoreValues`, but there is `ignoreKeywords`.

**Comment by jeddy3** [comment]
Thank you for pointing those out. I've edited [my comment](https://github.com/stylelint/stylelint/issues/4622#issuecomment-594641799) to correct the examples.

The [docs](https://github.com/stylelint/stylelint/tree/master/lib/rules/value-keyword-case) mention `ignoreValues`. I'll rustle up a quick pull request to address that.

**Comment by AlexWayfer** [comment]
Nice, thank you. Have a nice day.

**Comment by camiblanch** [comment]
I'm having an issue related to this where `ignoreProperties` only works if the property is all lowercase. It's not case sensitive. See #4644
